Sources/Architecture/X86/HardwareBreakpointManager.cpp

```diff
@@ -184,6 +184,29 @@ int HardwareBreakpointManager::getAvailableLocation() {
 
   return (it - _locations.begin());
 }
+
+ErrorCode HardwareBreakpointManager::isValid(Address const &address,
+                                             size_t size, Mode mode) const {
+  switch (size) {
+  case 1:
+    break;
+  case 8:
+    DS2LOG(Warning, "8-byte breakpoints not supported on all architectures");
```

Comment:
This is supported only for x86_64 targets, right? Can we check for the architecture of the process instead of always printing a warning?